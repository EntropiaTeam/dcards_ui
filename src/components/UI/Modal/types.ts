export type MaxWidth = 'lg' | 'sm' | undefined;

export type MaxWidthStyleProp = {
  maxWidth: MaxWidth;
};
