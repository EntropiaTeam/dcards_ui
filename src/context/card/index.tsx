import React, {
  createContext, useContext, useReducer, FC
} from 'react';
import { CardActionsType, UseCardContext } from './types/CardTypes';
import CardReducer, { initialState } from './reducer/CardReducer';
import { CardActions } from './actions/CardActions';

// Create Context
export const CardContext = createContext(initialState);
export const CardActionsContext = createContext({} as CardActionsType);
export const useCardContext = (): UseCardContext => (
  { state: useContext(CardContext), actions: useContext(CardActionsContext) }
);

export const CardProvider: FC = ({ children }) => {
  const [state, dispatch] = useReducer(CardReducer, initialState);

  // Actions
  const actions: CardActionsType = {
    setCardToEdit: CardActions.setCardToEdit(dispatch)
  };

  return (
    <CardContext.Provider value={state}>
      <CardActionsContext.Provider value={actions}>
        {children}
      </CardActionsContext.Provider>
    </CardContext.Provider>
  );
};
