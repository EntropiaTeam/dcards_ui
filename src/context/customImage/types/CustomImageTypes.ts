import {
  SET_CROPPER,
  SET_CROPPER_BACK_IMAGE_URL,
  SET_IMAGE_FILE,
  SET_CUSTOM_IMAGE_ERRORS,
  SET_TERM_AGREEMENT,
  SET_CROPPED_IMAGE,
  SET_CROPPED_CANVAS_DATA,
  SET_ROTATION_VALUE,
  SET_ZOOM_VALUE,
  RESET_CUSTOM_IMAGE_STATE,
  SET_AGREEMENT_TRANSLATION
} from '../constants/CustomImageConstants';

export type CropperCanvasData = {
  left: number;
  top: number;
  width: number;
  height: number;
  naturalWidth: number;
  naturalHeight: number;
};

export type CustomImageInitialState = {
  cropper: Cropper | null;
  cropperBackImageURL: string | null;
  userImage: string;
  croppedImage: Blob | null;
  cropperCanvasData: CropperCanvasData;
  rotationValue: number;
  zoomValue: number;
  errors: string[];
  agreement: boolean | null;
  agreementTranslation: Agreement | null;
};

export type Agreement = {
  agreement: {
    title: string;
    subTitle: string;
    updatedAtTitle: string;
    updatedAtDate: UpdatedAtDate;
    tos: string;
  };
};

export type UpdatedAtDate = {
  day: number;
  month: number;
  year: number;
};

export type AgreementTranslationType = (agreement: Agreement) => void;

export type CustomImageActionsType = {
  setCropper: (instance: Cropper | null) => void;
  setUserImage: (userImageBase64: string) => void;
  setCroppedImage: (croppedImageBlob: Blob | null) => void;
  setCroppedCanvasData: (canvasData: CropperCanvasData) => void;
  setCropperBackImageURL: (cropperBackImageURL: string | null) => void;
  setZoomRatio: (ratio: number) => void;
  setRotationDegree: (rotationValue: number) => void;
  setErrors: (errors: string[]) => void;
  setAgreement: (agreement: boolean) => void;
  setAgreementTranslation: AgreementTranslationType;
  resetCustomImageState: () => void;
};

// Context Type
export type UseCustomImageContext = {
  state: CustomImageInitialState;
  actions: CustomImageActionsType;
};

// Action creators' Types
export type SetCropperAction = {
  type: typeof SET_CROPPER;
  payload: Cropper | null;
};

export type SetUserImageAction = {
  type: typeof SET_IMAGE_FILE;
  payload: string;
};

export type SetZoomValueAction = {
  type: typeof SET_ZOOM_VALUE;
  payload: number;
};

export type SetRotationDegreeAction = {
  type: typeof SET_ROTATION_VALUE;
  payload: number;
};

export type SetErrorsAction = {
  type: typeof SET_CUSTOM_IMAGE_ERRORS;
  payload: string[];
};

export type SetTermAgreementAction = {
  type: typeof SET_TERM_AGREEMENT;
  payload: boolean;
};

export type SetAgreementTranslationAction = {
  type: typeof SET_AGREEMENT_TRANSLATION;
  payload: Agreement;
};

export type SetCroppedImageAction = {
  type: typeof SET_CROPPED_IMAGE;
  payload: Blob | null;
};

export type SetCropperBackImageURLAction = {
  type: typeof SET_CROPPER_BACK_IMAGE_URL;
  payload: string | null;
};

export type ResetCustomImageStateAction = {
  type: typeof RESET_CUSTOM_IMAGE_STATE;
};

export type SetCroppedCanvasDataAction = {
  type: typeof SET_CROPPED_CANVAS_DATA;
  payload: CropperCanvasData;
};

export type CustomImageActionTypes = SetCropperAction |
SetUserImageAction |
SetErrorsAction |
SetTermAgreementAction |
SetAgreementTranslationAction |
SetCroppedImageAction |
SetCropperBackImageURLAction |
SetCroppedCanvasDataAction |
SetRotationDegreeAction |
ResetCustomImageStateAction |
SetZoomValueAction;
