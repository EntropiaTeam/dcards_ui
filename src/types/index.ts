import { ReactCropperProps } from 'react-cropper';

export type EditorValidationConfig = {
  fullScreenMaxHeight: number;
  smScreenMaxHeight: number;
  lgScreenMaxHeight: number;
};

export type CardConfig = {
  id: string;
  name: string;
  frontImageUrl: string;
  innerImageUrl: string;
  textarea: {
    top: number;
    height: number;
  };
  sizes: {
    width: number;
    height: number;
    xsWidth: number;
    xsHeight: number;
  };
  font: {
    family: string;
    size: number;
    color: string;
    weight: number;
  };
  lineHeight: {
    sizeRatio: number;
  };
};

export type GlobalFontConfig = {
  family: string;
  src: string;
  weight: number;
};

export type ImageConfig = {
  width: number;
  height: number;
  xsWidth?: number;
  xsHeight?: number;
  rotation?: number;
  transform?: boolean;
  leftMargin?: number;
};

export type CropperConfig = ReactCropperProps;
