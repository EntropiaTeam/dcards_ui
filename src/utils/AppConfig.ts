import {
  ImageConfig, EditorValidationConfig, CardConfig, CropperConfig
} from '../types';
import { SingleCard } from '../context/card/types/CardTypes';
import { getImageUrl } from '../helpers';
import { Brand } from '../enums/Brand';

export interface AppConfiguration {
  customCardId: string;
  customCategoryId: string;
  maxTextareaLength: number;
  editorValidationConfig: EditorValidationConfig;
  cardConfig: CardConfig;
  cropperConfig: CropperConfig;
}

export class AppConfig {
  static [Brand.Edible] = {
    termsOfUseLink: 'https://www.ediblearrangements.com/legal/terms-of-use.aspx',
    privatePolicyLink: 'https://www.ediblearrangements.com/legal/privacy-policy.aspx'
  };

  static [Brand.FruitFlowers] = {
    termsOfUseLink: 'https://www.fruitflowers.com/termsandconditions',
    privatePolicyLink: 'https://www.fruitflowers.com/privacy'
  };

  customCardId: string;

  customCategoryId: string;

  maxTextareaLength: number;

  editorValidationConfig: EditorValidationConfig;

  card: CardConfig;

  cropper: CropperConfig;

  croppedImageConfig: Cropper.GetCroppedCanvasOptions;

  categoriesImgConfig: ImageConfig;

  cardsListImgConfig: ImageConfig;

  constructor() {
    this.customCardId = 'blank_photo_portrait';
    this.customCategoryId = 'custom_photo';
    this.maxTextareaLength = 1500;

    this.croppedImageConfig = {
      width: 1600,
      height: 2550,
      imageSmoothingQuality: 'high',
      fillColor: '#fff'
    };

    this.editorValidationConfig = {
      fullScreenMaxHeight: 170,
      smScreenMaxHeight: 97,
      lgScreenMaxHeight: 150
    };

    this.card = {
      id: '',
      name: '',
      frontImageUrl: '',
      innerImageUrl: '',
      sizes: {
        width: 302,
        height: 481,
        xsWidth: 246,
        xsHeight: 392
      },
      font: {
        family: 'Roboto Slab',
        size: 14,
        color: '#000',
        weight: 500
      },
      textarea: {
        top: 0,
        height: 0
      },
      lineHeight: {
        sizeRatio: 1.36
      }
    };

    this.cropper = {
      viewMode: 0,
      dragMode: 'move',
      autoCropArea: 1,
      restore: false,
      modal: false,
      guides: false,
      center: false,
      highlight: false,
      cropBoxMovable: false,
      cropBoxResizable: false,
      toggleDragModeOnDblclick: false,
      background: false,
      initialAspectRatio: 1600 / 2550,
      responsive: true,
      checkOrientation: false
    };

    this.categoriesImgConfig = {
      width: 105,
      height: 160,
      xsWidth: 92,
      xsHeight: 140,
      rotation: 0,
      transform: false,
      leftMargin: 0
    };

    this.cardsListImgConfig = {
      width: 250,
      height: 388,
      xsWidth: 157,
      xsHeight: 253
    };
  }

  public static percentsToPixelsHeight = (
    pageHeight: number, percent: number
  ): number => (percent / 100) * pageHeight;

  public initializeCardConfig(card: SingleCard, locale: string): void {
    this.customCardId = locale === 'fr-CA' ? 'FR_blank_photo_portrait' : 'blank_photo_portrait';
    this.card.id = card.id;
    this.card.frontImageUrl = getImageUrl(`${card.id}_front.png`, 'editor');
    this.card.innerImageUrl = getImageUrl(`${card.id}_inside.png`, 'editor');
    this.card.name = card.name;
    this.card.font = card.font;
    this.setTextareaParams(card);
  }

  public setTextareaParams(card: SingleCard): void {
    const defaultPadding = 2.8; // %
    const fullPageHeight = 2550; // (8.5" * 300dpi)
    const top = (card.textarea_offset && card.textarea_offset.top)
      ? card.textarea_offset.top : 0;
    const bottom = (card.textarea_offset && card.textarea_offset.bottom)
      ? card.textarea_offset.bottom : 0;

    const textareaTopPercent = ((top / fullPageHeight) * 100) + defaultPadding;
    const textareaBottomPercent = ((bottom / fullPageHeight) * 100) + defaultPadding;
    const height = 100 - textareaTopPercent - textareaBottomPercent;

    this.card.textarea.top = textareaTopPercent;
    this.card.textarea.height = height;
  }
}
