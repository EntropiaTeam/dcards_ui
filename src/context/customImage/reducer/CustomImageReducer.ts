import {
  SET_CUSTOM_IMAGE_ERRORS,
  SET_IMAGE_FILE,
  SET_CROPPER,
  SET_TERM_AGREEMENT,
  SET_CROPPED_IMAGE,
  SET_CROPPED_CANVAS_DATA,
  SET_ROTATION_VALUE,
  SET_ZOOM_VALUE,
  RESET_CUSTOM_IMAGE_STATE,
  SET_AGREEMENT_TRANSLATION,
  SET_CROPPER_BACK_IMAGE_URL
} from '../constants/CustomImageConstants';
import { CustomImageInitialState, CustomImageActionTypes } from '../types/CustomImageTypes';

// Initial state
export const initialState: CustomImageInitialState = {
  cropper: null,
  cropperBackImageURL: null,
  userImage: '',
  croppedImage: null,
  rotationValue: 0,
  zoomValue: 0,
  cropperCanvasData: {
    left: 0,
    top: 0,
    width: 0,
    height: 0,
    naturalWidth: 0,
    naturalHeight: 0
  },
  errors: [],
  agreement: null,
  agreementTranslation: null
};

export default (state = initialState, action: CustomImageActionTypes): CustomImageInitialState => {
  switch (action.type) {
    case SET_CROPPER:
      return {
        ...state,
        cropper: action.payload
      };
    case SET_CROPPER_BACK_IMAGE_URL:
      return {
        ...state,
        cropperBackImageURL: action.payload
      };
    case SET_CROPPED_CANVAS_DATA:
      return {
        ...state,
        cropperCanvasData: action.payload
      };
    case SET_IMAGE_FILE:
      return {
        ...state,
        userImage: action.payload
      };
    case SET_ROTATION_VALUE:
      return {
        ...state,
        rotationValue: action.payload
      };
    case SET_ZOOM_VALUE:
      return {
        ...state,
        zoomValue: action.payload
      };
    case SET_CROPPED_IMAGE:
      return {
        ...state,
        croppedImage: action.payload
      };

    case SET_CUSTOM_IMAGE_ERRORS:
      return {
        ...state,
        errors: action.payload
      };

    case SET_TERM_AGREEMENT:
      return {
        ...state,
        agreement: action.payload
      };

    case SET_AGREEMENT_TRANSLATION:
      return {
        ...state,
        agreementTranslation: action.payload
      };

    case RESET_CUSTOM_IMAGE_STATE:
      return {
        ...initialState,
        agreement: state.agreement,
        agreementTranslation: state.agreementTranslation
      };

    default:
      return state;
  }
};
