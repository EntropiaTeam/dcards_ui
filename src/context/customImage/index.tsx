import React, {
  createContext, useContext, useReducer, useMemo, FC
} from 'react';
import { CustomImageActionsType, UseCustomImageContext } from './types/CustomImageTypes';
import { CustomImageActions } from './actions/CustomImageActions';
import CustomImageReducer, { initialState } from './reducer/CustomImageReducer';

// Create Context
export const CustomImageContext = createContext(initialState);
export const CustomImageActionsContext = createContext({} as CustomImageActionsType);
export const useCustomImageContext = (): UseCustomImageContext => (
  { state: useContext(CustomImageContext), actions: useContext(CustomImageActionsContext) }
);

export const CustomImageProvider: FC = ({ children }) => {
  const [state, dispatch] = useReducer(CustomImageReducer, initialState);
  // Actions
  const actions: CustomImageActionsType = useMemo(() => ({
    setErrors: CustomImageActions.setErrors(dispatch),
    setUserImage: CustomImageActions.setUserImage(dispatch),
    setCropper: CustomImageActions.setCropper(dispatch),
    setAgreement: CustomImageActions.setAgreement(dispatch),
    setAgreementTranslation: CustomImageActions.setAgreementTranslation(dispatch),
    setCroppedImage: CustomImageActions.setCroppedImageValue(dispatch),
    setCropperBackImageURL: CustomImageActions.setCropperBackImageValue(dispatch),
    setRotationDegree: CustomImageActions.setRotationDegree(dispatch),
    setZoomRatio: CustomImageActions.setZoomRatio(dispatch),
    setCroppedCanvasData: CustomImageActions.setCroppedCanvasData(dispatch),
    resetCustomImageState: CustomImageActions.resetCustomImageState(dispatch)
  }), [dispatch]);

  return (
    <CustomImageContext.Provider value={state}>
      <CustomImageActionsContext.Provider value={actions}>
        {children}
      </CustomImageActionsContext.Provider>
    </CustomImageContext.Provider>
  );
};
