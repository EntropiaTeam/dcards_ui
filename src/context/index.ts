import { CardProvider, useCardContext } from './card';
import { CategoriesProvider, useCategoriesContext } from './categories';
import { TextProvider, useTextContext } from './text';
import { OrderProvider, useOrderContext } from './order';
import { CustomImageProvider, useCustomImageContext } from './customImage';
import { NavigationProvider, useNavigationContext } from './navigation';

export {
  CardProvider,
  useCardContext,
  CategoriesProvider,
  useCategoriesContext,
  TextProvider,
  useTextContext,
  OrderProvider,
  useOrderContext,
  CustomImageProvider,
  useCustomImageContext,
  NavigationProvider,
  useNavigationContext
};
