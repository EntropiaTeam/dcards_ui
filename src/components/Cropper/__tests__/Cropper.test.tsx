import React from 'react';
import renderer from 'react-test-renderer';
import CropperContainer from '../index';
import { CustomImageContext, CustomImageActionsContext } from '../../../context/customImage';
import { initialState as customImageInitialState } from '../../../context/customImage/reducer/CustomImageReducer';
import { CustomImageInitialState } from '../../../context/customImage/types/CustomImageTypes';
import { AppConfig } from '../../../utils/AppConfig';

interface IntlMock {
  useTranslation: () => void;
}

interface TranslationMock {
  t: () => void;
}

jest.mock('react-i18next', (): IntlMock => ({
  useTranslation: (): TranslationMock => ({
    t: (key: string): string => key
  })
}));

describe('<CropperContainer />', () => {
  const config = new AppConfig();

  const frontImageUrl = 'https://steu2sprntdevstore01.blob.core.windows.net/assets/i/Halloween_A_editor_front.png';

  it('should render CropperControls component if there is no errors', () => {
    const state: CustomImageInitialState = {
      ...customImageInitialState,
      userImage: 'base64:image'
    };
    const cropperRenderer = renderer.create(
      <CustomImageContext.Provider value={state}>
        <CustomImageActionsContext.Provider value={{ setCropper: () => null }}>
          <CropperContainer
            cropperConfig={config.cropper}
            cardConfig={config.card}
            imageUrl={frontImageUrl}
          />
        </CustomImageActionsContext.Provider>
      </CustomImageContext.Provider>
    );
    expect(cropperRenderer.toJSON()).toMatchSnapshot();
  });
});
