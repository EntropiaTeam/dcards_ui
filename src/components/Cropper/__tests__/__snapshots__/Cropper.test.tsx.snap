// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CropperContainer /> should render CropperControls component if there is no errors 1`] = `
<div
  className="MuiBox-root MuiBox-root-13 makeStyles-cropperContainer-1"
>
  <div
    className="MuiBox-root MuiBox-root-14 makeStyles-sliderContainer-2"
  >
    <button
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-15 MuiButton-text makeStyles-zoomIconBtn-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={3}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="40"
            rx="2"
            width="40"
          />
          <path
            d="M24.7143 22.8286L31 29.1143L29.1143 31L22.8286 24.7143V23.7211L22.4891 23.3691C21.056 24.6011 19.1954 25.3429 17.1714 25.3429C15.0042 25.3429 12.9258 24.4819 11.3934 22.9495C9.86092 21.4171 9 19.3386 9 17.1714C9 15.0042 9.86092 12.9258 11.3934 11.3934C12.9258 9.86092 15.0042 9 17.1714 9C19.3386 9 21.4171 9.86092 22.9495 11.3934C24.4819 12.9258 25.3429 15.0042 25.3429 17.1714C25.3429 19.1954 24.6011 21.056 23.3691 22.4891L23.7211 22.8286H24.7143ZM17.1714 22.8286C20.3143 22.8286 22.8286 20.3143 22.8286 17.1714C22.8286 14.0286 20.3143 11.5143 17.1714 11.5143C14.0286 11.5143 11.5143 14.0286 11.5143 17.1714C11.5143 20.3143 14.0286 22.8286 17.1714 22.8286ZM20.3143 17.8H17.8V20.3143H16.5429V17.8H14.0286V16.5429H16.5429V14.0286H17.8V16.5429H20.3143V17.8Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
    <div
      className="MuiBox-root MuiBox-root-16 makeStyles-slider-3"
    >
      <span
        className="MuiSlider-root MuiSlider-colorPrimary MuiSlider-vertical WithStyles(ForwardRef(Slider))-vertical-17"
        onMouseDown={[Function]}
        tabIndex={-1}
      >
        <span
          className="MuiSlider-rail WithStyles(ForwardRef(Slider))-rail-20"
        />
        <span
          className="MuiSlider-track WithStyles(ForwardRef(Slider))-track-19"
          style={
            Object {
              "bottom": "0%",
              "height": "0%",
            }
          }
        />
        <input
          type="hidden"
          value="0.05"
        />
        <span
          aria-orientation="vertical"
          aria-valuemax={1.1}
          aria-valuemin={0.05}
          aria-valuenow={0.05}
          className="MuiSlider-thumb WithStyles(ForwardRef(Slider))-thumb-18 MuiSlider-thumbColorPrimary"
          data-index={0}
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseLeave={[Function]}
          onMouseOver={[Function]}
          role="slider"
          style={
            Object {
              "bottom": "0%",
            }
          }
          tabIndex={0}
        />
      </span>
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-15 MuiButton-text makeStyles-zoomIconBtn-4"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={4}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          fill="none"
          height="40"
          viewBox="0 0 40 40"
          width="40"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="none"
            height="40"
            rx="2"
            width="40"
          />
          <path
            d="M24.7143 22.8286H23.7211L23.3691 22.4891C24.6011 21.056 25.3429 19.1954 25.3429 17.1714C25.3429 15.0042 24.4819 12.9258 22.9495 11.3934C21.4171 9.86092 19.3386 9 17.1714 9C15.0042 9 12.9258 9.86092 11.3934 11.3934C9.86092 12.9258 9 15.0042 9 17.1714C9 19.3386 9.86092 21.4171 11.3934 22.9495C12.9258 24.4819 15.0042 25.3429 17.1714 25.3429C19.1954 25.3429 21.056 24.6011 22.4891 23.3691L22.8286 23.7211V24.7143L29.1143 31L31 29.1143L24.7143 22.8286ZM17.1714 22.8286C14.0286 22.8286 11.5143 20.3143 11.5143 17.1714C11.5143 14.0286 14.0286 11.5143 17.1714 11.5143C20.3143 11.5143 22.8286 14.0286 22.8286 17.1714C22.8286 20.3143 20.3143 22.8286 17.1714 22.8286ZM14.0286 16.5429H20.3143V17.8H14.0286V16.5429Z"
            fill="currentColor"
          />
        </svg>
      </span>
    </button>
  </div>
  <div
    className="MuiBox-root MuiBox-root-28 makeStyles-imageContainer-26 makeStyles-imageContainer-27 makeStyles-imgContainer-5"
    tabIndex={-1}
  >
    <div
      className="sc-bdVaJa GvOTx"
    >
      <img
        alt="picture"
        src="base64:image"
        style={
          Object {
            "maxWidth": "100%",
            "opacity": 0,
          }
        }
      />
    </div>
  </div>
  <div
    className="MuiBox-root MuiBox-root-30 makeStyles-actionBtns-29"
  >
    <button
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-15 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={5}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          fill="none"
          height="30"
          viewBox="0 0 30 30"
          width="30"
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="currentColor"
            height="16"
            rx="2"
            width="16"
            x="6"
            y="9.5"
          />
          <path
            d="M26 16V8C26 6.34315 24.6569 5 23 5H15"
            stroke="currentColor"
          />
          <path
            d="M17.5 2L14.5 5L17.5 8"
            stroke="currentColor"
          />
        </svg>
      </span>
    </button>
    <button
      className="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-15 MuiButton-text"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={6}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          fill="none"
          height={30}
          viewBox="0 0 30 30"
          width={30}
          xmlns="http://www.w3.org/2000/svg"
        >
          <rect
            fill="currentColor"
            height="2"
            rx="1"
            width="16"
            x="7"
            y="6"
          />
          <rect
            fill="currentColor"
            height="2"
            rx="1"
            width="4"
            x="13"
            y="5"
          />
          <path
            clipRule="evenodd"
            d="M7.14051 10.124C7.0659 9.52718 7.53129 9 8.13278 9H21.8672C22.4687 9 22.9341 9.52718 22.8595 10.124L21.1095 24.124C21.0469 24.6245 20.6215 25 20.1172 25H9.88278C9.37846 25 8.95306 24.6245 8.89051 24.124L7.14051 10.124ZM12 13C12 12.4477 12.4477 12 13 12C13.5523 12 14 12.4477 14 13V21C14 21.5523 13.5523 22 13 22C12.4477 22 12 21.5523 12 21V13ZM17 12C16.4477 12 16 12.4477 16 13V21C16 21.5523 16.4477 22 17 22C17.5523 22 18 21.5523 18 21V13C18 12.4477 17.5523 12 17 12Z"
            fill="currentColor"
            fillRule="evenodd"
          />
        </svg>
      </span>
    </button>
  </div>
</div>
`;
